gammapy/utils/fitting/tests/test_iminuit.py

```diff
@@ -0,0 +1,25 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+from numpy.testing import assert_allclose
+from ...modeling import ParameterList, Parameter
+from ...testing import requires_dependency
+from .. import fit_minuit
+
+
+@requires_dependency('iminuit')  
+def test_iminuit():
+    def f(parameters):
+        x = parameters['x'].value
+        y = parameters['y'].value
+        z = parameters['z'].value
+        return (x - 2) ** 2 + (y - 3) ** 2 + (z - 4) ** 2
+
+    p_in = ParameterList(
+        [Parameter('x', 2.1), Parameter('y', 3.1), Parameter('z', 4.1)]
+    )
+
+    p_out = fit_minuit(function=f, parameters=p_in)
```

Comment:
Should p_in change? Make a copy and leave unchanged better?
In any case please add assert to establish behavior via the test.

Reply:
Good point, I ran into :beetle: s that were caused by changing input parameters several times, so I decided to make a copy internally for now. Of course this can change later